src/Lean/Linter/Util.lean

```diff
@@ -20,6 +20,29 @@ do
   let messages := (← get).messages |>.add (mkMessageCore ctx.fileName ctx.fileMap content severity range.start range.stop)
   modify ({ · with messages := messages })
 
+def collectMacroExpansions? {m} [Monad m] (range : String.Range) (tree : Elab.InfoTree) : m <| Option <| List Elab.MacroExpansionInfo := do
```

Comment:
`go` returns `some _` if the targeted syntax range is in the given node or its children. If instead of `some []` it would return `none`, then nodes higher up in the InfoTree wouldn't know which list returned by their children is the one to prepend a `MacroExpansionInfo` to (`filterMap id` does this selection).

`some []` is returned to indicate that the correct InfoTree node was found, but there were no `MacroExpansionInfo`s on the way up yet.

Reply:
Could you add a docstring here?

Reply:
:+1: